relays/parachains/src/parachains_loop.rs

```diff
@@ -129,48 +96,47 @@ pub trait TargetClient<P: ParachainsPipeline>: RelayClient {
 	/// Get best block id.
 	async fn best_block(&self) -> Result<HeaderIdOf<P::TargetChain>, Self::Error>;
 
-	/// Get best finalized source block id.
-	async fn best_finalized_source_block(
+	/// Get best finalized source relay chain block id.
+	async fn best_finalized_source_relay_chain_block(
 		&self,
 		at_block: &HeaderIdOf<P::TargetChain>,
-	) -> Result<HeaderIdOf<P::SourceChain>, Self::Error>;
+	) -> Result<HeaderIdOf<P::SourceRelayChain>, Self::Error>;
 
-	/// Get parachain head hash at given block.
-	///
-	/// The implementation may call `ParachainsLoopMetrics::update_best_parachain_block_at_target`
-	/// on provided `metrics` object to update corresponding metric value.
+	/// Get parachain head id at given block.
 	async fn parachain_head(
 		&self,
 		at_block: HeaderIdOf<P::TargetChain>,
-		metrics: Option<&ParachainsLoopMetrics>,
-		para_id: ParaId,
-	) -> Result<Option<BestParaHeadHash>, Self::Error>;
+	) -> Result<Option<HeaderIdOf<P::SourceParachain>>, Self::Error>;
 
 	/// Submit parachain heads proof.
-	async fn submit_parachain_heads_proof(
+	async fn submit_parachain_head_proof(
 		&self,
-		at_source_block: HeaderIdOf<P::SourceChain>,
-		updated_parachains: Vec<(ParaId, ParaHash)>,
+		at_source_block: HeaderIdOf<P::SourceRelayChain>,
+		para_head_hash: ParaHash,
 		proof: ParaHeadsProof,
 	) -> Result<Self::TransactionTracker, Self::Error>;
 }
 
 /// Return prefix that will be used by default to expose Prometheus metrics of the parachains
 /// sync loop.
 pub fn metrics_prefix<P: ParachainsPipeline>() -> String {
-	format!("{}_to_{}_Parachains", P::SourceChain::NAME, P::TargetChain::NAME)
+	format!(
+		"{}_to_{}_Parachains_{}",
```

Comment:
Nit: Would it make sense to use something like:
```
	format!(
		"{}_Parachain_{}_to_{}",
		P::SourceRelayChain::NAME,
		P::SourceParachain::PARACHAIN_ID,
		P::TargetChain::NAME,
	)
``` 
?

Reply:
Actually I'd prefer similar relays (if we'll ever have it) to have similar prefix - i.e. `Rialto_to_Millau_Parachains_1000` and `Rialto_to_Millau_Parachains_2000` vs `Rialto_Parachain_1000_to_Millau` and `Rialto_Parachain_2000_to_Millau`. But let's revisit it later